bin/proving-service/src/commands/proxy.rs

```diff
@@ -45,10 +45,26 @@ impl StartProxy {
     /// - The server cannot be started.
     #[tracing::instrument(target = MIDEN_PROVING_SERVICE, name = "proxy:execute")]
     pub async fn execute(&self) -> Result<(), String> {
+        // Check if all required ports are available
+        check_port_availability(&self.proxy_config.host, self.proxy_config.port, "Proxy")?;
+        check_port_availability(
+            &self.proxy_config.host,
+            self.proxy_config.workers_update_port,
+            "Workers update",
+        )?;
+
+        if self.proxy_config.metrics_config.enable_metrics {
+            check_port_availability(
+                &self.proxy_config.metrics_config.prometheus_host,
+                self.proxy_config.metrics_config.prometheus_port,
+                "Metrics",
+            )?;
```

Comment:
I believe this is an external port that our service is connecting to or no? Or do we "serve" the metrics? If so then there should be no issue with duplicated ports.

Reply:
We use the prometheus HTTP server that Pingora provides to serve an endpoint that reports Prometheus metrics collected in the entire service